crates/primitives/src/transaction/mod.rs

```diff
@@ -820,6 +820,188 @@ impl Encodable for Transaction {
     }
 }
 
+impl alloy_consensus::Transaction for Transaction {
```

Comment:
I'd like to do this separately, this is definitely useful, but I'd like to do this as a separate change